common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -1019,6 +1019,12 @@ def check_problem(self, data, override_time=False):
 
         _ = self.runtime.service(self, "i18n").ugettext
 
+        # Add information from the related asides to the event
+        for aside in self.runtime.get_asides(self):
+            for field_key, field_val in aside.fields.iteritems():
+                if field_val.is_set_on(aside):
+                    event_info[field_key] = field_val.read_from(aside)
```

Comment:
Approach-wise, maybe it would be better to call into the aside to ask it to add (or return) any data it wants to associate with the event? That way a) the event data isn't openended, and b) the control over data representation rests with the Aside